niri-config/tests/wiki-parses.rs

```diff
@@ -0,0 +1,108 @@
+use std::fs;
+use std::path::PathBuf;
+
+const NO_TEST_COMMENT: &str = "<!-- no-test -->";
+
+struct KdlCodeBlock {
+    filename: String,
+    code: String,
+    line_number: usize,
+}
+
+fn extract_kdl_from_file(file_contents: &str, filename: &str) -> Vec<KdlCodeBlock> {
+    // Removes the > from callouts that might contain ```kdl```
+    let lines: Vec<_> = file_contents
+        .lines()
+        .map(|line| {
+            let line = line.trim();
+            if line.starts_with(">") {
+                if line.len() == 1 {
+                    ""
+                } else {
+                    &line[2..]
+                }
+            } else {
+                line
+            }
+        })
+        .enumerate()
+        .collect();
+
+    let mut lines_iter = lines.iter();
```

Comment:
I think you might be able to skip the collect/iter and use it directly?

Reply:
I tried to. Due to me having to check the previous line for a `<!-- no-test -->` it's impossible (or would lead to worse code)

Reply:
Ah, I see